src/brsTypes/components/RoSGNode.ts

```diff
@@ -498,6 +499,19 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         },
     });
 
+    /** Returns true if the field exists */
+    private hasfield = new Callable("hasfield", {
+        signature: {
+            args: [new StdlibArgument("fieldname", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, fieldname: BrsString) => {
+            return this.fields.has(fieldname.value.toLowerCase())
```

Comment:
For future reference, this could be as simple as:

```suggestion
            return BrsBoolean.from(
                this.fields.has(fieldname.value.toLowerCase())
            )
```

What you've got is of course fine though 😄